test/e2e/main/main_test.go

```diff
@@ -195,7 +200,9 @@ func TestNodeStartup(t *testing.T) {
 	defer os.Unsetenv(v1.OperatorPodTargetLabelsEnvVarName)
 	// Register it
 	testKube.CreateInfinispan(spec, tutils.Namespace)
-	defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
+	spec.Labels = map[string]string{"test-name": t.Name()}
```

Comment:
Label should be defined before Infinispan create as `spec.Labels["test-name"] = t.Name()` as labels already utilized in this test